### ISSUE TITLE
Infinite loop due uncontrolled offset increment at ProductPartitionTree class

### ISSUE THREAD

**Issue by morci7** [googleads/googleads-java-lib#56]
Hi,

I've found when we use method "createAdGroupTree" at ProductPartitionTree class never finish due bad offset increment.

```java
...
                offset += adGroupCriterionPage.getEntries().length;
                selectorBuilder.increaseOffsetBy(offset);
            }
        } while(offset < adGroupCriterionPage.getTotalNumEntries().intValue());
...
```

Seams offset is incremented 2 times. We should remove this line:

```java
...
                offset += adGroupCriterionPage.getEntries().length;
...
```

**Comment by jradcliff** [comment]
Sorry - we actually fixed this in the [v201502](https://github.com/googleads/googleads-java-lib/blob/master/modules/adwords_axis/src/main/java/com/google/api/ads/adwords/axis/utils/v201502/shopping/ProductPartitionTree.java#L227) and [v201506](https://github.com/googleads/googleads-java-lib/blob/master/modules/adwords_axis/src/main/java/com/google/api/ads/adwords/axis/utils/v201506/shopping/ProductPartitionTree.java#L227) versions of `ProductPartitionTree` (see [issue #50]), but it looks like it slipped back in on the [v201509 version](https://github.com/googleads/googleads-java-lib/blob/master/modules/adwords_axis/src/main/java/com/google/api/ads/adwords/axis/utils/v201509/shopping/ProductPartitionTree.java#L227).

I'll fix the v201509 in an upcoming patch. Thanks for pointing this out.

**Comment by jradcliff** [comment]
Fixed in [2.7.0](https://github.com/googleads/googleads-java-lib/releases/tag/2.7.0) and added tests to ensure this does not resurface.
